### PR TITLE
Issue with ks.merge to Series

### LINKED ISSUE #817: Bug in merge dataframe to series.
```python
df = ks.DataFrame([['a', 1,'dog'], ['b', 2,'cat']],
                    columns=['letter', 'number','animal'])
s1 = ks.Series(['a', 'e'],index=[0,1],name='letter')
df.merge(s1,how='outer')
```
The above code return this error: 
```python
>>> AttributeError: Field columns not found, possible values are __index_level_0__, letter
```

### PR DESCRIPTION
**Pull request by thoo**
Fix #817 

Also add pytests for merging `ks.DataFrame` to `ks.Series`.

### CODE REVIEW

**Comment by thoo** [comment]
The reason Travis CI is failing is `pandas.merge` added Series objects in version 0.24.0 and one of the tests was run with pandas version 0.23.

File: databricks/koalas/tests/test_dataframe.py [hunk 1]
```diff
@@ -764,6 +766,23 @@ def check(op):
...
+
+        # Test join types with Series
+        for how in ['inner', 'left', 'right', 'outer']:
+            check(lambda left, right: left.merge(right, how=how), right_ks, right_ps)
+            check(lambda left, right: left.merge(right, left_on='x', right_on='x', how=how),
+                  right_ks, right_ps)
+
+        # suffix
+        check(lambda left, right: left.merge(right, suffixes=['_left', '_right'], how='outer',
+                                             left_index=True, right_index=True), right_ks, right_ps)
```
Comment (ueshin) [inline databricks/koalas/tests/test_dataframe.py]: Let's skip tests when pandas is older than 0.24. See the example with `LooseVersion` in `test_groupby.py`.

File: databricks/koalas/tests/test_dataframe.py [hunk 2]
```diff
@@ -767,21 +768,23 @@ def check(op, right_kdf=right_kdf, right_pdf=right_pdf):
...
-              right_ks, right_ps)
-        check(lambda left, right: left.set_index('x').merge(right, left_index=True, right_on='x'),
-              right_ks, right_ps)
-
-        # Test join types with Series
-        for how in ['inner', 'left', 'right', 'outer']:
-            check(lambda left, right: left.merge(right, how=how), right_ks, right_ps)
-            check(lambda left, right: left.merge(right, left_on='x', right_on='x', how=how),
+        # pd.DataFrame.merge with Series is implemented since version 0.24.0
+        if LooseVersion(pd.__version__) > LooseVersion("0.24.2"):
```
Comment (ueshin) [inline databricks/koalas/tests/test_dataframe.py]: nit: shall we use `0.24.0` if pandas supports since then?

Comment (thoo) [reply databricks/koalas/tests/test_dataframe.py]: Sure.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**